Engine/ac/dynobj/scriptstring.cpp

```diff
@@ -35,8 +36,10 @@ const char *ScriptString::GetType() {
 }
 
 int ScriptString::Serialize(const char *address, char *buffer, int bufsize) {
-    if (text == NULL)
-        text = "";
+    if (text == nullptr) {
+        text = static_cast<char *>(malloc(1));
```

Comment:
wouldn't this cause a memory leak ?

Reply:
Nope, we free it in Dispose. The problem was that we were trying to free a string literal.

Reply:
Hmm, I guess ScriptString with null pointer inside was never a case...
But frankly, I think this should be fixed differently, changing function logic instead to something like:
<pre>
    size_t len = text == nullptr ? 0 : strlen(text);
    SerializeInt(len);
    if (text)
        strcpy(&serbuffer[bytesSoFar], text);
    bytesSoFar += len + 1;
</pre>

Reply:
I did something like that. Main difference was that you need to still strcpy or add a null char even if text len is 0.  (to match up with the unserialize implementation)